src/components/tools/NewToolRequestDialog.js

```diff
@@ -0,0 +1,180 @@
+import React, { useState } from "react";
+
+import { useTranslation } from "i18n";
+import { useMutation } from "react-query";
+
+import { Field, Form, Formik } from "formik";
+
+import DEDialog from "components/utils/DEDialog";
+import ErrorTypographyWithDialog from "components/utils/error/ErrorTypographyWithDialog";
+import ids from "./ids";
+import constants from "../../constants";
+import { toolRequest } from "serviceFacades/tools";
+
+import {
+    build,
+    announce,
+    FormMultilineTextField,
+    FormTextField,
+} from "@cyverse-de/ui-lib";
+
+import Button from "@material-ui/core/Button";
+
+export default function NewToolRequestDialog(props) {
+    const { open, baseId, onClose } = props;
+    const [requestError, setRequestError] = useState();
+
+    const { t } = useTranslation("tools");
+
+    const [submitNewToolRequest, { status: submitRequestStatus }] = useMutation(
+        ({ submission }) => toolRequest(submission),
+        {
+            onSuccess: (data) => {
+                announce({
+                    text: t("toolRequestSuccess"),
+                });
+                onClose();
+            },
+            onError: setRequestError,
+        }
+    );
+
+    const handleSubmit = (values, { props }) => {
+        const submission = { ...values };
+        //test_data_path and cmd_line are required by service. Most of the times those info. can be
+        // found in the source repo. So copy the source url to those fields if not provided by user.
+        if (!values.test_data_path) {
+            submission.test_data_path = values.source_url;
+        }
+        if (!values.cmd_line) {
+            submission.cmd_line = values.source_url;
+        }
+        if (submitRequestStatus !== constants.LOADING) {
+            submitNewToolRequest({ submission });
+        }
+    };
+
+    const validateUrl = (value) => {
+        let error;
+        if (!constants.URL_REGEX.test(value)) {
+            error = t("validationErrMsgURL");
+        }
+        return error;
+    };
+
+    return (
+        <DEDialog
+            open={open}
+            onClose={onClose}
+            baseId={build(baseId, ids.TOOL_REQUEST.DIALOG)}
+            title={t("newToolRequestDialogHeading")}
+        >
+            {requestError && (
+                <ErrorTypographyWithDialog
+                    errorObject={requestError}
+                    errorMessage={t("toolRequestError")}
+                    baseId={baseId}
+                />
+            )}
+            <Formik
+                initialValues={{
+                    name: "",
+                    description: "",
+                    source_url: "",
+                    version: "",
+                    documentation_url: "",
+                    cmd_line: "",
+                    test_data_path: "",
+                }}
+                enableReinitialize={true}
+                onSubmit={handleSubmit}
+            >
+                {() => {
+                    return (
```

Comment:
I think if you're not using any of Formik's helper props, you can begin the `<Form>` tag as a child to `<Formik>`, meaning you don't need this extra function and return